File: sdk/systeminfo/DeviceAPI.php
<?php
namespace systeminfo;

/**
 * Class DeviceAPI
 * @package systeminfo
 */
class DeviceAPI
{
    /**
     * --RU--
     * Получить имя устройства
     * --EN--
     * Get device name
     * @return string
     */
    public static function getDeviceName() {}

    /**
     * --RU--
     * Получить название операционной системы
     * --EN--
     * Get OS name
     * @return string
     */
    public static function getOsName() {}

    /**
     * --RU--
     * Получить версию операционной системы
     * --EN--
     * Get OS version
     * @return string
     */
    public static function getOsVersion() {}

    /**
     * --RU--
     * Получить имя пользователя
     * --EN--
     * Get user name
     * @return string
     */
    public static function getUserName() {}

    /**
     * --RU--
     * Получить MAC-адрес устройства
     * --EN--
     * Get MAC address
     * @return string
     */
    public static function getMacAddress() {}

    /**
     * --RU--
     * Получить IP-адрес устройства
     * --EN--
     * Get IP address
     * @return string
     */
    public static function getIpAddress() {}

    /**
     * --RU--
     * Получить список сетевых интерфейсов
     * --EN--
     * Get network interfaces
     * @return string
     */
    public static function getNetworkInterfaces() {}

    /**
     * --RU--
     * Получить время работы системы в секундах
     * --EN--
     * Get uptime
     * @return long
     */
    public static function getUptime() {}

    /**
     * --RU--
     * Получить архитектуру устройства
     * --EN--
     * Get architecture
     * @return string
     */
    public static function getArchitecture() {}
}

File: src-jvm/main/java/org/develnext/jphp/ext/systeminfo/classes/DeviceAPI.java
package org.develnext.jphp.ext.systeminfo.classes;

import org.develnext.jphp.ext.systeminfo.SystemAPIExtension;
import php.runtime.annotation.Reflection;
import php.runtime.annotation.Reflection.Signature;
import php.runtime.env.Environment;
import php.runtime.lang.BaseObject;
import php.runtime.reflection.ClassEntity;
import oshi.SystemInfo;
import oshi.hardware.HardwareAbstractionLayer;
import oshi.hardware.NetworkIF;
import oshi.software.os.OperatingSystem;

import java.net.InetAddress;
import java.net.NetworkInterface;
import java.net.SocketException;
import java.net.UnknownHostException;
import java.util.List;

@Reflection.Namespace(SystemAPIExtension.NS)
public class DeviceAPI extends BaseObject {
    private static final SystemInfo systemInfo = new SystemInfo();
    private static final HardwareAbstractionLayer hal = systemInfo.getHardware();
    private static final OperatingSystem os = systemInfo.getOperatingSystem();

    public DeviceAPI(Environment env) {
        super(env);
    }

    protected DeviceAPI(ClassEntity entity) {
        super(entity);
    }

    public DeviceAPI(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Signature
    public static String getDeviceName() throws UnknownHostException {
        InetAddress addr = InetAddress.getLocalHost();
        return addr.getHostName();
    }

    @Signature
    public static String getOsName() {
        return System.getProperty("os.name");
    }

    @Signature
    public static String getOsVersion() {
        return os.getVersionInfo().getVersion();
    }

    @Signature
    public static String getUserName() {
        return System.getProperty("user.name");
    }

    @Signature
    public static String getMacAddress() throws SocketException, UnknownHostException {
        InetAddress addr = InetAddress.getLocalHost();
        NetworkInterface ni = NetworkInterface.getByInetAddress(addr);
        byte[] mac = ni.getHardwareAddress();
        if (mac != null) {
            StringBuilder sb = new StringBuilder();
            for (byte b : mac) {
                sb.append(String.format("%02X:", b));
            }
            return sb.toString().substring(0, sb.length() - 1);
        } else {
            return "Unknown";
        }
    }

    @Signature
    public static String getIpAddress() throws UnknownHostException {
        InetAddress addr = InetAddress.getLocalHost();
        return addr.getHostAddress();
    }

    @Signature
    public static String getNetworkInterfaces() {
        StringBuilder sb = new StringBuilder();
        List<NetworkIF> networkIFs = hal.getNetworkIFs();
        for (NetworkIF net : networkIFs) {
            sb.append(net.getName()).append("\n");
        }
        return sb.toString();
    }

    @Signature
    public static long getUptime() {
        return os.getSystemUptime();
    }

    @Signature
    public static String getArchitecture() {
        return System.getProperty("os.arch");
    }
}
